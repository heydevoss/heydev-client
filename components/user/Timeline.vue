<template>
  <div class="block">
    <br /><el-timeline>
      <el-timeline-item
        v-for="(activity, index) in activities"
        :key="index"
        :icon="activity.icon"
        :type="activity.type"
        :color="activity.color"
        :size="activity.size"
        :timestamp="activity.timestamp"
      >
        {{ activity.content }}
      </el-timeline-item>
    </el-timeline>
  </div>
</template>

<script>
export default {
  data() {
    return {
      activities: [
        {
          content: 'New Pull Request on Tamburetei',
          timestamp: '2018-04-12 20:46',
          size: 'large',
          type: 'primary',
          color: 'grey',
          icon: 'el-icon-s-promotion'
        },
        {
          content: 'Commented on IssueAi #48 issue',
          timestamp: '2018-04-03 20:46',
          color: 'grey',
          icon: 'el-icon-more',
          size: 'large'
        },
        {
          content: 'Star in Roadmap',
          timestamp: '2018-04-03 20:46',
          size: 'large',
          icon: 'el-icon-star-on',
          color: 'grey'
        },
        {
          content: 'Commit on IssueAi',
          timestamp: '2018-04-03 20:46',
          icon: 'el-icon-circle-plus-outline',
          color: 'grey',
          size: 'large'
        }
      ]
    }
  }
}
</script>
