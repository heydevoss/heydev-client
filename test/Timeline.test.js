import { shallowMount } from '@vue/test-utils'
import Timeline from '@/components/user/Timeline'

describe('<Timeline/> Component', () => {
  const wrapper = shallowMount(Timeline)

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
