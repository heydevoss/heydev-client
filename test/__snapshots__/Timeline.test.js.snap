// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline/> Component renders correctly 1`] = `
<div
  class="block"
>
  <br />
  <el-timeline>
    <el-timeline-item
      color="grey"
      icon="el-icon-s-promotion"
      size="large"
      timestamp="2018-04-12 20:46"
      type="primary"
    >
      
        New Pull Request on Tamburetei
        
    </el-timeline-item>
    <el-timeline-item
      color="grey"
      icon="el-icon-more"
      size="large"
      timestamp="2018-04-03 20:46"
    >
      
        Commented on IssueAi #48 issue
        
    </el-timeline-item>
    <el-timeline-item
      color="grey"
      icon="el-icon-star-on"
      size="large"
      timestamp="2018-04-03 20:46"
    >
      
        Star in Roadmap
        
    </el-timeline-item>
    <el-timeline-item
      color="grey"
      icon="el-icon-circle-plus-outline"
      size="large"
      timestamp="2018-04-03 20:46"
    >
      
        Commit on IssueAi
        
    </el-timeline-item>
  </el-timeline>
</div>
`;
